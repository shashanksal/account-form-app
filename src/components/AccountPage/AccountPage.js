import React from 'react';
import { useSelector } from 'react-redux';

import { Grid, Typography } from '@material-ui/core';
import { createStyles, makeStyles } from '@material-ui/core/styles'; // Using Material UI styles

import { dateConvert } from '../../utils/helper';

const useStyles = makeStyles((theme) =>
	createStyles({
		paper: {
			padding: theme.spacing(2),
		},
		titleFont: {
			fontWeight: 600,
			fontSize: 14,
			color: '#05AE95',
		},
		bodyFont: {
			fontSize: 14,
		},
	})
);

const AccountPage = (props) => {
	const user = useSelector((state) => state.rAccount);
	console.log(new Date(user.dob).toDateString());
	const classes = useStyles();
	return (
		<div>
			<Grid container spacing={2}>
				<Grid item sm={6}>
					<Typography
						display="block"
						align="left"
						color="textPrimary"
						className={classes.titleFont}
					>
						First Name
					</Typography>
					<Typography align="left" className={classes.bodyFont}>
						{user.first_name}
					</Typography>
				</Grid>
				<Grid item sm={6}>
					<Typography
						display="block"
						align="left"
						className={classes.titleFont}
					>
						{user.last_name}
					</Typography>
					<Typography align="left" className={classes.bodyFont}>
						Banner
					</Typography>
				</Grid>
			</Grid>
			<Grid container spacing={2}>
				<Grid item sm={12}>
					<Typography
						display="block"
						align="left"
						color="textPrimary"
						className={classes.titleFont}
					>
						Email
					</Typography>
					<Typography align="left" className={classes.bodyFont}>
						{user.email}
					</Typography>
				</Grid>
			</Grid>
			<Grid container spacing={2}>
				<Grid item sm={12}>
					<Typography
						display="block"
						align="left"
						color="textPrimary"
						className={classes.titleFont}
					>
						Phone
					</Typography>
					<Typography align="left" className={classes.bodyFont}>
						{user.phone}
					</Typography>
				</Grid>
			</Grid>
			<Grid container spacing={2}>
				<Grid item sm={12}>
					<Typography
						display="block"
						align="left"
						color="textPrimary"
						className={classes.titleFont}
					>
						Date of Birth
					</Typography>
					<Typography align="left" className={classes.bodyFont}>
						{dateConvert(user.dob)}
					</Typography>
				</Grid>
			</Grid>
			<Grid container spacing={2}>
				<Grid item sm={12}>
					<Typography
						display="block"
						align="left"
						color="textPrimary"
						className={classes.titleFont}
					>
						Bio
					</Typography>
					<Typography align="left" className={classes.bodyFont}>
						{user.bio}
					</Typography>
				</Grid>
			</Grid>
		</div>
	);
};

export default AccountPage;
