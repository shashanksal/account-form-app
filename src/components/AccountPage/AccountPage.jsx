import React from "react";
import { useSelector } from "react-redux";

import { Grid, Typography } from "@material-ui/core";
import { createStyles, makeStyles } from "@material-ui/core/styles"; // Using Material UI styles

import { dateConvert } from "../../utils/helper";

const useStyles = makeStyles((theme) =>
	createStyles({
		paper: {
			padding: theme.spacing(2)
		},
		lableFont: {
			fontWeight: 600,
			fontSize: 14,
			color: "#05AE95"
		},
		bodyFont: {
			fontSize: 14
		}
	})
);

const AccountPage = (props) => {
	const user = useSelector((state) => state?.rAccount);
	const classes = useStyles();
	return (
		<div>
			<Grid container spacing={2}>
				<Grid item sm={6}>
					<Typography
						display="block"
						align="left"
						color="textPrimary"
						className={classes.lableFont}
					>
						First Name
					</Typography>
					<Typography align="left" className={classes.bodyFont}>
						{user?.first_name}
					</Typography>
				</Grid>
				<Grid item sm={6}>
					<Typography
						display="block"
						align="left"
						className={classes.lableFont}
					>
						Last Name
					</Typography>
					<Typography align="left" className={classes.bodyFont}>
						{user?.last_name}
					</Typography>
				</Grid>
			</Grid>
			<Grid container spacing={2}>
				<Grid item sm={12}>
					<Typography
						display="block"
						align="left"
						color="textPrimary"
						className={classes.lableFont}
					>
						Email
					</Typography>
					<Typography align="left" className={classes.bodyFont}>
						{user?.email}
					</Typography>
				</Grid>
			</Grid>
			<Grid container spacing={2}>
				<Grid item sm={12}>
					<Typography
						display="block"
						align="left"
						color="textPrimary"
						className={classes.lableFont}
					>
						Phone
					</Typography>
					<Typography align="left" className={classes.bodyFont}>
						{user?.phone}
					</Typography>
				</Grid>
			</Grid>
			<Grid container spacing={2}>
				<Grid item sm={12}>
					<Typography
						display="block"
						align="left"
						color="textPrimary"
						className={classes.lableFont}
					>
						Date of Birth
					</Typography>
					<Typography align="left" className={classes.bodyFont}>
						{dateConvert(user?.dob)}
					</Typography>
				</Grid>
			</Grid>
			<Grid container spacing={2}>
				<Grid item sm={12}>
					<Typography
						display="block"
						align="left"
						color="textPrimary"
						className={classes.lableFont}
					>
						Bio
					</Typography>
					<Typography align="left" className={classes.bodyFont}>
						{user?.bio}
					</Typography>
				</Grid>
			</Grid>
		</div>
	);
};

export default AccountPage;
