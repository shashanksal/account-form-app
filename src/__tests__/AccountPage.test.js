import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

import AccountPage from "../components/AccountPage";

const middlewares = [];
const mockStore = configureMockStore(middlewares);

const initData = {};

const mStore = mockStore(initData);

describe("test AccountPage", () => {
	it("AccountPage matches snapshot", () => {
		const { asFragment } = render(
			<Provider store={mStore}>
				<AccountPage />
			</Provider>
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
