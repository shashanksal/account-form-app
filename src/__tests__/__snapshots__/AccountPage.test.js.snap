// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test AccountPage AccountPage matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-6"
      >
        <p
          class="MuiTypography-root makeStyles-lableFont-2 MuiTypography-body1 MuiTypography-colorTextPrimary MuiTypography-alignLeft MuiTypography-displayBlock"
        >
          First Name
        </p>
        <p
          class="MuiTypography-root makeStyles-bodyFont-3 MuiTypography-body1 MuiTypography-alignLeft"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-6"
      >
        <p
          class="MuiTypography-root makeStyles-lableFont-2 MuiTypography-body1 MuiTypography-alignLeft MuiTypography-displayBlock"
        />
        <p
          class="MuiTypography-root makeStyles-bodyFont-3 MuiTypography-body1 MuiTypography-alignLeft"
        >
          Banner
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12"
      >
        <p
          class="MuiTypography-root makeStyles-lableFont-2 MuiTypography-body1 MuiTypography-colorTextPrimary MuiTypography-alignLeft MuiTypography-displayBlock"
        >
          Email
        </p>
        <p
          class="MuiTypography-root makeStyles-bodyFont-3 MuiTypography-body1 MuiTypography-alignLeft"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12"
      >
        <p
          class="MuiTypography-root makeStyles-lableFont-2 MuiTypography-body1 MuiTypography-colorTextPrimary MuiTypography-alignLeft MuiTypography-displayBlock"
        >
          Phone
        </p>
        <p
          class="MuiTypography-root makeStyles-bodyFont-3 MuiTypography-body1 MuiTypography-alignLeft"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12"
      >
        <p
          class="MuiTypography-root makeStyles-lableFont-2 MuiTypography-body1 MuiTypography-colorTextPrimary MuiTypography-alignLeft MuiTypography-displayBlock"
        >
          Date of Birth
        </p>
        <p
          class="MuiTypography-root makeStyles-bodyFont-3 MuiTypography-body1 MuiTypography-alignLeft"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12"
      >
        <p
          class="MuiTypography-root makeStyles-lableFont-2 MuiTypography-body1 MuiTypography-colorTextPrimary MuiTypography-alignLeft MuiTypography-displayBlock"
        >
          Bio
        </p>
        <p
          class="MuiTypography-root makeStyles-bodyFont-3 MuiTypography-body1 MuiTypography-alignLeft"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
